internal/pkg/agent/application/actions/handlers/handler_action_policy_change.go

```diff
@@ -220,25 +224,80 @@ func (h *PolicyChangeHandler) handleFleetServerHosts(ctx context.Context, c *con
 	return nil
 }
 
-func clientEqual(k1 remote.Config, k2 remote.Config) bool {
-	if k1.Protocol != k2.Protocol {
+// applyConfigWithPrecedence applies Proxy and TLS configs, but ignores empty or
+// absent ones.
+// That way a proxy or TLS config set during install/enroll using cli flags
+// won't be overridden by an absent or empty proxy from fleet-server.
+// However, if there is a proxy or TLS config sent by fleet-server, it'll take
+// precedence. Therefore, it's not possible to remove a proxy or TLS config once
+// it's set.
+func (h *PolicyChangeHandler) applyConfigWithPrecedence(cfg remote.Config) {
+	defaultcfg := configuration.DefaultFleetAgentConfig()
+
+	if cfg.Protocol != defaultcfg.Client.Protocol ||
+		cfg.Host != defaultcfg.Client.Host ||
+		!slices.Equal(cfg.Hosts, defaultcfg.Client.Hosts) ||
+		cfg.Path != defaultcfg.Client.Path {
+		h.config.Fleet.Client.Protocol = cfg.Protocol
+		h.config.Fleet.Client.Path = cfg.Path
+		h.config.Fleet.Client.Host = cfg.Host
+		h.config.Fleet.Client.Hosts = cfg.Hosts
+	}
+
+	if cfg.Transport.Proxy.URL == nil ||
+		cfg.Transport.Proxy.URL.String() == "" {
+		h.log.Debug("proxy from fleet is empty or null, the proxy will not be changed")
+	} else {
+		h.config.Fleet.Client.Transport.Proxy = cfg.Transport.Proxy
+		h.log.Debug("received proxy from fleet, applying it")
+	}
+
+	emptyCertificate := tlscommon.CertificateConfig{}
+	if cfg.Transport.TLS != nil {
+		if h.config.Fleet.Client.Transport.TLS == nil {
+			h.config.Fleet.Client.Transport.TLS = &tlscommon.Config{}
+		}
+
+		// client certificate
+		if cfg.Transport.TLS.Certificate == emptyCertificate {
+			h.log.Debug("TLS certificate/key from fleet are empty or null, the TLS config will not be changed")
+		} else {
+			h.config.Fleet.Client.Transport.TLS.Certificate = cfg.Transport.TLS.Certificate
+			h.log.Debug("received TSL certificate/key from fleet, applying it")
+		}
+
+		// CA
+		if len(cfg.Transport.TLS.CAs) == 0 {
+			h.log.Debug("TLS CAs from fleet are empty or null, the TLS config will not be changed")
+		} else {
+			h.config.Fleet.Client.Transport.TLS.CAs = cfg.Transport.TLS.CAs
+			h.log.Debug("received SSL from fleet, applying it")
+		}
+	}
+}
+
+func clientEqual(current remote.Config, new remote.Config) bool {
+	if new.Protocol != "" &&
+		current.Protocol != new.Protocol {
 		return false
 	}
-	if k1.Path != k2.Path {
+	if new.Path != "" &&
+		current.Path != new.Path {
 		return false
 	}
 
-	sort.Strings(k1.Hosts)
-	sort.Strings(k2.Hosts)
-	if len(k1.Hosts) != len(k2.Hosts) {
+	sort.Strings(current.Hosts)
+	sort.Strings(new.Hosts)
+	if len(current.Hosts) != len(new.Hosts) {
 		return false
 	}
-	for i, v := range k1.Hosts {
-		if v != k2.Hosts[i] {
+	for i, v := range current.Hosts {
+		if v != new.Hosts[i] {
 			return false
 		}
 	}
 
+	// should it ignore empty headers?
```

Comment:
Empty headers from Fleet? This should have the same precedence rules as everything else.